database-commons/src/main/java/io/cdap/plugin/util/DBUtils.java

```diff
@@ -212,10 +213,7 @@ public static Schema getSchema(int sqlType, int precision, int scale) throws SQL
 
       case Types.NUMERIC:
       case Types.DECIMAL:
-        // if there are no digits after the point, use integer types
-        type = scale != 0 ? Schema.Type.DOUBLE :
-          // with 10 digits we can represent 2^32 and LONG is required
-          precision > 9 ? Schema.Type.LONG : Schema.Type.INT;
+        type = Schema.Type.BYTES;
```

Comment:
how come this isn't returning a decimal?

Reply:
It certainly must return `Schema.decimalOf(precision, scale)` just like in `CommonSchemaReader#getSchema`, thanks. 
Fixed.

Reply:
does this mean this code path wasn't getting tested in the unit tests?

Reply:
Decimal logical type is based on the `Schema.Type.BYTES` primitive type: 
https://github.com/cdapio/cdap/blob/develop/cdap-api-common/src/main/java/io/cdap/cdap/api/data/schema/Schema.java#L124
```
DECIMAL(Schema.Type.BYTES, "decimal");
```

Thus it's ok to use `Schema.Type.BYTES` in that case and unit tests didn't help to find this.

Reply:
Also we have to ensure that we test sink plugins with explicit input schema set. I will add these tests in the following PRs for all data types support.